apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -362,14 +454,15 @@ namespace MWRender
             cullingMode |= osg::CullStack::SMALL_FEATURE_CULLING;
         }
 
-        mViewer->getCamera()->setCullingMode( cullingMode );
-
-        mViewer->getCamera()->setComputeNearFarMode(osg::Camera::DO_NOT_COMPUTE_NEAR_FAR);
```

Comment:
Why can't we just remove this line and have OSG use the computed near and far planes? I don't get why we need a clamp projection matrix callback.

Reply:
near plane might be far compare to a hardcoded one...it allows to narrow depth range as it can be

Reply:
You said

> it's literrally a copy/paste from default osg ClampProjectionMatrixImplementation, nothing to elaborate, just clamping near plane in a callback mean to be used for that

so I don't get why we're not just relying on that implementation.

Reply:
the ClampProjection matrix  callback is called at each leaf added to the stategraph.
it allows to give yourself constraints on the desired projection matrix
(like min near plane or max far plane)
more important, using bounding volume culling the cullvisitor use this clamped perspective matrix to maintain the frustum polytope...

Reply:
the near plane clamping is the only code change